srcs/server/src/game/game.service.ts

```diff
@@ -0,0 +1,821 @@
+    // Add it to the game and set his side
+    player.socket.leave(Params.LOBBY);
+    player.socket.join(game.id);
+    game.players.push({
+      ...player,
+      side: side,
+      score: 0,
+    });
+    return game;
+  }
+
+  /** Check if a player is already in a game */
+  private _isPlayerInGame(userId: number): boolean {
+    if (
+      this.games.find((game) =>
+        game.players.find((player) => player.userId === userId),
+      )
+    )
+      return true;
+    return false;
+  }
+
+  /** Check if a viewer is already in a game */
+  private _isViewerInGame(userId: number): boolean {
+    if (
+      this.games.find((game) =>
+        game.viewers.find((viewer) => viewer.userId === userId),
+      )
+    )
+      return true;
+    return false;
+  }
+
+  /** Get the side of a player in a game from his id */
+  private _getSideFromUser(game: Game, userId: number): number {
+    const player: Player = game.players.find(
+      (player) => player.userId === userId,
+    );
+    if (!player) throw new PlayerNotFoundException(userId);
+    return player.side;
+  }
+
+  /** Create the game list from the global server data */
+  private _createGameList(): object {
+    const gameList = this.games.map((game) => {
+      return {
+        id: game.id,
+        players: game.players.map((p) => ({
+          userId: p.userId,
+          pic: p.pic,
+          name: p.name,
+          score: p.score,
+        })),
+        viewersCount: game.viewers.length,
+      };
+    });
+    return gameList;
+  }
+
+  /** Cancel game */
+  private _cancelGame(game: Game) {
+    this.games = this.games.filter((g) => g.id !== game.id);
+    this.server.in(game.id).socketsJoin(Params.LOBBY);
+    this.server.in(game.id).socketsLeave(game.id);
+    // send a fresh gamelist to the lobby
+    const gameList = this._createGameList();
+    this.server.to(Params.LOBBY).emit('gameList', gameList);
+  }
+
+  /** End a game */
+  private async _endGame(
+    game: Game,
+    motive: number,
+    loserId?: number,
+  ): Promise<Match> {
+    // remove the game from the list
+    this.games = this.games.filter((g) => g.id !== game.id);
+    // emit a game end info to all players / viewers so they go back to lobby
+    this.server.to(game.id).emit('gameEnd', motive);
+    // disconnect players / viewers from game room and connect them to lobby
+    this.server.in(game.id).socketsJoin(Params.LOBBY);
+    this.server.in(game.id).socketsLeave(game.id);
+    // send a fresh gamelist to the lobby
+    const gameList = this._createGameList();
+    this.server.to(Params.LOBBY).emit('gameList', gameList);
+    // save game result in database
+    const createMatchDto: CreateMatchDto = {
+      players: game.players.map((p) => ({
+        playerId: p.userId,
+        side: p.side,
+        score: p.score,
+        status: p.userId === loserId ? (motive === Motive.ABANDON ? 2 : 1) : 0,
+      })),
+    };
+    try {
+      return await this.matchService.create(createMatchDto);
```

Comment:
no need to await this no ? because we are not doing anything afterwards

Reply:
I prefer to always wait for async function even in that case. It can be important for testing purposes for exemple.